agent/src/main/java/com/teamscale/jacoco/agent/upload/artifactory/ArtifactoryConfig.java

```diff
@@ -0,0 +1,170 @@
+				return true;
+			default:
+				return false;
+		}
+	}
+
+	/** Checks if none of the required fields is null. */
+	public boolean hasAllRequiredFieldsSet() {
+		return url != null && user != null && password != null;
+	}
+
+	/** Checks if all required fields are null. */
+	public boolean hasAllRequiredFieldsNull() {
+		return url == null && user == null && password == null;
+	}
+
+	/** Checks whether commit and revision are set. */
+	public boolean hasCommitInfo() {
+		return commitInfo != null;
+	}
+
+	/** Parses the commit information form a git.properties file. */
+	public static CommitInfo parseGitProperties(FilePatternResolver filePatternResolver,
+												DateTimeFormatter gitPropertiesCommitTimeFormat, String optionName,
+												String value) throws AgentOptionParseException {
+		File jarFile = filePatternResolver.parsePath(optionName, value).toFile();
+		try {
+			CommitInfo commitInfo = parseGitProperties(jarFile, gitPropertiesCommitTimeFormat);
+			if (commitInfo == null) {
+				throw new InvalidGitPropertiesException(
+						"Could not locate a git.properties file in " + jarFile.toString());
+			}
+			return commitInfo;
+		} catch (IOException | InvalidGitPropertiesException e) {
+			throw new AgentOptionParseException("Could not locate a valid git.properties file in " + jarFile.toString(),
+					e);
+		}
+	}
+
+	/** Parses the commit information form a git.properties file. */
+	public static CommitInfo parseGitProperties(File jarFile,
+												DateTimeFormatter gitPropertiesCommitTimeFormat) throws IOException, InvalidGitPropertiesException {
+		Pair<String, Properties> entryWithProperties = GitPropertiesLocator.findGitPropertiesInJar(jarFile);
+		if (entryWithProperties == null) {
+			return null;
+		}
+
+		String entry = entryWithProperties.getFirst();
+		Properties properties = entryWithProperties.getSecond();
+
+		String revision = GitPropertiesLocator
+				.getGitPropertiesValue(properties, GitPropertiesLocator.GIT_PROPERTIES_GIT_COMMIT_ID, entry,
+						jarFile);
+		String branchName = GitPropertiesLocator
+				.getGitPropertiesValue(properties, GitPropertiesLocator.GIT_PROPERTIES_GIT_BRANCH, entry, jarFile);
+		long timestamp = ZonedDateTime.parse(GitPropertiesLocator
+						.getGitPropertiesValue(properties, GitPropertiesLocator.GIT_PROPERTIES_GIT_COMMIT_TIME, entry,
+								jarFile),
+				gitPropertiesCommitTimeFormat).toInstant().toEpochMilli();
+		return new CommitInfo(revision, new CommitDescriptor(branchName, timestamp));
+	}
+
+	/** Hold information regarding a commit. */
+	public static class CommitInfo {
+		/** The revision information (git hash). */
+		public String revision;
+
+		/** The commit descriptor. */
+		public CommitDescriptor commit;
+
+		/** Constructor. */
+		public CommitInfo(String revision, CommitDescriptor commit) {
+			this.revision = revision;
+			this.commit = commit;
+		}
+
+		@Override
+		public String toString() {
+			return commit + "/" + revision;
```

Comment:
Should we add a comment here where this is used?

Can we include "Commit "+commit+"/Revision "+revision or is this the path in artifactory?